Chargebee/Classes/Configuration/CBEnvironment.swift

```diff
@@ -42,5 +42,35 @@ class CBEnvironment {
         }
 
     }
+    
+    func configure(site: String, apiKey: String, allowErrorLogging: Bool, sdkKey: String? = nil, handler: @escaping CBAuthenticationHandler) {
+            CBEnvironment.site = site
+            CBEnvironment.apiKey = apiKey
+            CBEnvironment.allowErrorLogging = allowErrorLogging
+            CBEnvironment.encodedApiKey = CBEnvironment.apiKey.data(using: .utf8)?.base64EncodedString() ?? ""
+            CBEnvironment.baseUrl = "https://\(CBEnvironment.site).chargebee.com/api"
+            CBEnvironment.version = .unknown
+
+            let (onSuccess, onError) = CBResult.buildResultHandlers(handler, nil)
+
+            if let sdkKey = sdkKey {
+                CBEnvironment.sdkKey = sdkKey
+                // Verify SDK Key and Setup the Environment
+                CBAuthenticationManager().authenticate(forSDKKey: CBEnvironment.sdkKey) { result in
+                    switch result {
+                    case .success(let status):
+                        print("Environment Setup - Completed")
+                        print("Note: pre-requisites configuration is mandatory for SDK to work. Learn more(https://www.chargebee.com/docs/2.0/mobile-app-store-product-iap.html)")
+                        CBEnvironment.version = status.details.version ?? .unknown
+                        onSuccess(status)
+                    case .error(let error):
+                        print(error)
+                        print("Note: pre-requisites configuration is mandatory for SDK to work. Learn more(https://www.chargebee.com/docs/2.0/mobile-app-store-product-iap.html)")
+                        CBEnvironment.version = .unknown
+                        onError(error)
+                    }
+                }
+            }
+        }
```

Comment:
Can we extract the common functional code among these methods instead of duplicating?

Reply:
thanks @cb-haripriyan, its done!